src/Polly.Shared/Retry/RetryTResultSyntax.cs

```diff
@@ -241,6 +240,37 @@ public static RetryPolicy<TResult> WaitAndRetry<TResult>(this PolicyBuilder<TRes
             );
         }
 
+        /// <summary>
+        /// Builds a <see cref="Policy"/> that will wait and retry.
+        /// calling <paramref name="onRetry"/> on each retry with the handled exception or result and the current sleep duration.
+        /// On each retry, the duration to wait is calculated by calling <paramref name="sleepDurationStrategy"/> with
+        /// the current retry number (1 for first retry, 2 for second etc).
+        /// </summary>
+        /// <param name="policyBuilder">The policy builder.</param>
+        /// <param name="retryCount">The retry count.</param>
+        /// <param name="sleepDurationStrategy">The class that provides the duration to wait for for a particular retry attempt.</param>
+        /// <param name="onRetry">The action to call on each retry.</param>
+        /// <returns>The policy instance.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">retryCount;Value must be greater than or equal to zero.</exception>
+        /// <exception cref="System.ArgumentNullException">
+        /// sleepDurationProvider
+        /// or
+        /// onRetry
+        /// </exception>
+        public static RetryPolicy<TResult> WaitAndRetry<TResult>(this PolicyBuilder<TResult> policyBuilder, int retryCount, ISleepDurationStrategy sleepDurationStrategy, Action<DelegateResult<TResult>, TimeSpan> onRetry)
```

Comment:
This overload uses `ISleepDurationStrategy` instead of `IEnumerable<TimeSpan>`

Reply:
Note that we don't have to implement any of these functions yet. We can provide the strategies and let consumers use them directly to generate `TimeSpan[]` series with which they can call the `IEnumerable<TimeSpan>` overloads.

Reply:
Per decision in previous comment, we won't be integrating new overloads for now